lib/temporal/testing/replay_tester.rb

```diff
@@ -0,0 +1,136 @@
+require 'gen/temporal/api/history/v1/message_pb'
+require 'json'
+require 'temporal/errors'
+require 'temporal/metadata/workflow_task'
+require 'temporal/middleware/chain'
+require 'temporal/workflow/executor'
+require 'temporal/workflow/stack_trace_tracker'
+
+module Temporal
+  module Testing
+    class ReplayError < StandardError
+    end
+
+    class ReplayTester
+      def initialize(config: Temporal.configuration)
+        # Duplicate the configuration so that this doesn't interfere with other tests in the
+        # same process that are not replay tests
+        @config = config.dup.tap do |c|
+          # Otherwise, replay tests will produce no logs. This helps with test debugging.
+          c.log_on_workflow_replay = true
+        end
+      end
+
+      # Runs a replay test by loading a file from the given file path that contains JSON. This JSON can be
+      # downloaded using the .get_workflow_history_json method on the Temporal client, through the Temporal CLI,
+      # or the Temporal UI.
+      #
+      # If the replay test succeeds, the method will return silently. If the replay tests fails, an error will be raised.
+      #
+      # Common problems and solutions:
+      # - "Unexpected event UNSPECIFIED". This can occur if the history was downloaded with Java-style enum values, which
+      #   can happen in certain versions of certain SDKs. Use the correct_event_types function below to convert these.
+      #   See that function's comment for more details.
+      def replay_history_json_file(workflow_class, path)
+        json = File.read(path)
+        replay_history_json(workflow_class, json)
+      end
+
+      # Runs a replay test on a JSON string directly. See comment on replay_history_json_file more details.
+      def replay_history_json(workflow_class, json)
```

Comment:
Feels a bit more composable to have just one typesafe `replay_history` method which takes a `History` and then have some helper methods to convert from proto/json/files to `History`
Was perusing the [python](https://github.com/temporalio/sdk-python/blob/58d695106a339e9e0be6fc02c78a4dcf57ce6eca/temporalio/worker/_replayer.py#L96) version, might be worth looking for insights there as well.

Reply:
It's not as close as it could be because temporal-ruby already has its own representation of workflow histories that I didn't want to refactor, but I was able to bring its more composable API shape over here